File: 2285. Maximum Total Importance of Roads.go
func maximumImportance(n int, roads [][]int) int64 {
    var answer, importance int64 = 0, 1
    degrees := make([]int, n)
    for _, road := range roads {
        degrees[road[0]]++
        degrees[road[1]]++
    }
    sort.Ints(degrees)
    for _, d := range degrees {
        answer += int64(d) * importance
        importance++
    }
    return answer
}

File: 2285. Maximum Total Importance of Roads.java
class Solution {
    public long maximumImportance(int n, int[][] roads) {
        long answer = 0, importance = 1;
        int[] degrees = new int[n];
        for (int[] road : roads) {
            degrees[road[0]]++;
            degrees[road[1]]++;
        }
        Arrays.sort(degrees);
        for (int d : degrees) answer += (importance++) * d;
        return answer;
    }
}
